rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/TechnologyIdentified.java

```diff
@@ -0,0 +1,146 @@
+package org.jboss.windup.rules.apps.javaee;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation;
+import org.jboss.windup.graph.model.FileLocationModel;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.graph.model.WindupVertexFrame;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.reporting.model.TagSetModel;
+import org.jboss.windup.reporting.service.TagSetService;
+import org.jboss.windup.rules.apps.javaee.model.stats.TechnologyUsageStatisticsModel;
+import org.jboss.windup.util.Logging;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ * Adds the specified statistics and tag information regarding an technology that has been located in the analyzed application.
+ *
+ * If no count is specified, then a default of "1" is assumed.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class TechnologyIdentified extends AbstractIterationOperation<WindupVertexFrame>
+            implements TechnologyIdentifiedWithCount, TechnologyIdentifiedWithName
```

Comment:
How can this implement `TechnologyIdentifiedWithCount` and `TechnologyIdentifiedWithName` if both of these 2 interfaces have incompatible methods `withTag` and don't extend each other?

```java
public interface TechnologyIdentifiedWithName extends OperationBuilder
{
    TechnologyIdentifiedWithName withTag(String tag);
    TechnologyIdentifiedWithCount numberFound(int count);
}
```
```java
public interface TechnologyIdentifiedWithCount {
    TechnologyIdentifiedWithCount withTag(String tag);
}
```

(I got it, these methods return instance of `TechnologyIdentified` which implements both of these interfaces, but shouldn't there be some relation between them? Maybe one should extend other?)

Btw aren't names of these interfaces opposite? I would expect the `TechnologyIdentifiedWithCount` to have some method with count (`numberFound(int count);`) 

Also the `TechnologyIdentifiedWithName` should have something to do with name. It only has `withTag` so maybe it should be `TechnologyIdentifiedWithTag` ?

Reply:
This is in keeping with the pattern used by JavaClass, Hint, and other fluent APIs in our codebase.

I am fine with moving away from this pattern, but I kept it here. The naming conventions for the classes follows those naming conventions as well. We can discuss it at the standup if people would like to change that, though.